x/programs/cmd/simulator/cmd/storage.go

```diff
@@ -54,31 +59,76 @@ func (s *prefixedStateMutable) Remove(ctx context.Context, key []byte) error {
 }
 
 func newAccountPrefixedMutable(account codec.Address, mutable state.Mutable) state.Mutable {
-	return &prefixedStateMutable{inner: mutable, prefix: programStateKey(account[:])}
+	return &prefixedStateMutable{inner: mutable, prefix: accountStateKey(account[:])}
 }
 
 //
 // Program
 //
 
-func programStateKey(key []byte) (k []byte) {
+func accountStateKey(key []byte) (k []byte) {
+	k = make([]byte, 2+len(key))
+	k[0] = accountPrefix
+	copy(k[1:], key)
+	k[len(key)-1] = accountStatePrefix
```

Comment:
This function rewrites the 3rd byte from the end, is your intention to write the last one ? You could maybe write
```suggestion
	k[len(key)+1] = accountStatePrefix
```
You could add a unit test in `storage_test.go`

```go
package cmd

import (
	"testing"

	"github.com/stretchr/testify/require"
)

func TestPrefix(t *testing.T) {
	require := require.New(t)
	key := []byte{0, 1, 2, 3}
	stateKey := accountStateKey(key)
	require.Equal(len(stateKey), len(key) + 2)
	expectedStateKey := append([]byte{accountPrefix}, key...)
	expectedStateKey = append(expectedStateKey, accountStatePrefix)
	require.Equal(stateKey, expectedStateKey)
}
```